### PR TITLE
[FLINK-10368][e2e] Hardened kerberized yarn e2e test

### PR DESCRIPTION
**Pull request by dawidwys**
Hardened the kerberized yarn e2e test, especially having in mind environments poor in resources.

Changes:

 * wait for whole  bootstrapping script to execute on master node before submitting job
 * added check that all containers are up and running before submitting job
 * reduced memory requirements for the kerberized yarn test

### CODE REVIEW

File: flink-end-to-end-tests/test-scripts/test_yarn_kerberos_docker.sh [hunk 1]
```diff
@@ -60,19 +64,41 @@ function cluster_shutdown {
...
-    sleep 5
+# wait for kerberos to be set up
+start_time=$(date +%s)
+until docker logs master 2>&1 | grep -q "Finished master initialization"; do
+    current_time=$(date +%s)
+    time_diff=$((current_time - start_time))
+
+    if [ $time_diff -ge $MAX_RETRY_SECONDS ]; then
+        echo "ERROR: Could not start hadoop cluster. Aborting..."
+        exit 0
```
Comment (aljoscha) [inline flink-end-to-end-tests/test-scripts/test_yarn_kerberos_docker.sh]: Isn't exit code `1` the exit code for failure?

File: flink-end-to-end-tests/test-scripts/test_yarn_kerberos_docker.sh [hunk 2]
```diff
@@ -60,19 +64,41 @@ function cluster_shutdown {
...
 done
 
+# perform health checks
+if ! { [ $(docker inspect -f '{{.State.Running}}' master 2>&1) = 'true' ] &&
+       [ $(docker inspect -f '{{.State.Running}}' slave1 2>&1) = 'true' ] &&
+       [ $(docker inspect -f '{{.State.Running}}' slave2 2>&1) = 'true' ] &&
+       [ $(docker inspect -f '{{.State.Running}}' kdc 2>&1) = 'true' ]; };
+then
+    echo "ERROR: Could not start hadoop cluster. At least one of the containers failed. Aborting..."
+    exit 0
```
Comment (aljoscha) [inline flink-end-to-end-tests/test-scripts/test_yarn_kerberos_docker.sh]: Isn't exit code `1` the exit code for failure?

**Comment by dawidwys** [comment]
Yes, you are right, I've used this error code to gracefully abort the test. Wasn't it the suggested idea in JIRA ticket? 

I wouldn't say it should happen often. I've tried running the test with resources limited for docker containers and it generally passed. I don't know which is the better, I can see both pros & cons of both approaches.

**Comment by aljoscha** [comment]
Ah, you're right! It was even my who said that Till proposed this. If @tillrohrmann is alright with this approach I'd say this is good to merge.

**Comment by tillrohrmann** [comment]
Would it work to retry the deployment logic and fail if we could not deploy the cluster after several attempts?

**Comment by dawidwys** [comment]
At first I thought it wouldn't change much (as the cluster starting consists only of bringing up docker containers), but actually there is some kerberos setting up logic, so it might actually make sense.

Just to be clear we should fail the test meaning (exit 1).

**Comment by tillrohrmann** [comment]
That sounds promising. Let us know once you've updated the PR.

**Comment by dawidwys** [comment]
@tillrohrmann @aljoscha I've updated the test. Could you have final look?

**Comment by aljoscha** [comment]
I think this looks good to merge now. :👌

**Pull request merged**